CONTRIBUTING.md

```diff
@@ -132,25 +140,25 @@ project:
    git checkout -b <topic-branch-name>
    ```
 
-4. Commit your changes in logical chunks. Please adhere to these [git commit
-   message guidelines](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
-   or your code is unlikely be merged into the main project. Use Git's
-   [interactive rebase](https://help.github.com/articles/interactive-rebase)
-   feature to tidy up your commits before making them public.
+4. Please commit your changes in logical chunks, and try to provide clear commit
+   messages.
 
-5. We have a handy [checklist for new igraph
-   functions](https://github.com/igraph/igraph/wiki/Checklist-for-new-(and-old)-functions).
+5. We have a [checklist for new igraph functions](https://github.com/igraph/igraph/wiki/Checklist-for-new-(and-old)-functions).
    If you have added any new functions to igraph, please go through the
    checklist to ensure that your functions play nicely with the rest of the
    library.
 
-6. Locally merge (or rebase) the upstream development branch into your topic branch:
+6. Make sure that your PR is based off the latest code and locally merge (or
+   rebase) the upstream development branch into your topic branch:
 
    ```bash
    git pull [--rebase] upstream <dev-branch>
    ```
 
-7. Push your topic branch up to your fork:
+7. If necessary, you can use git's
```

Comment:
Same comment as for point 4.

Rebase is dangerous, and one can accidentally ruin one's work with it. I have. Do we want this in the beginner instructions?

Reply:
Do you mean only the _interactive_ rebasing or the `--rebase` option in general from point 6?

I agree that interactive rebasing is advanced stuff; even I use it only in rare occasions. However, rebasing when pulling is probably useful as the alternative is merging, in which case you need to deal with merge conflicts. And I think there is always an easy way out of `git pull --rebase upstream` by running `git rebase --abort`.

Reply:
Yes, I mean the interactive rebasing. I have messed up my work when doing interactive rebasing, encountering conflicts and failing to clean them up correctly. All I wanted was a tidy commit history (point 4). I think we should not encourage that in this document.

Reply:
> However, rebasing when pulling is probably useful

Yes, I have no problem with that. That is useful to mention.